### ISSUE TITLE
Convert Layer1D to Layer2D and  back

### ISSUE THREAD

**Issue by kmun** [LGSInnovations/sigplot#97]
**Is your feature request related to a problem? Please describe.**
I would like to change a Layer2D static spectral plot (waterfall plot) that is currently being displayed on the screen to a set of individual colored traces, i.e. if the waterfall plot has 8 rows, create an 8 trace plot (either flat or 3d, doesn't matter  to me).  I know how to do it in software, but I was wondering if you can do it interactively on the screen, i.e. the user can switch between the two plotting formats via the built in menu.  I don't see it as an option, but maybe I'm missing something.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by maihde** [comment]
@kmun are you familiar with the xCut and yCut behavior of Layer2D?  If not, see if that works for you.  Basically if you hover the mouse over a position on a raster and press X it will make a 1D representation of the x (or y) axis at that location.  You then press X again to leave the cut.

**Comment by maihde** [comment]
The other option, if you know how to accomplish it in code, is to make your own key-binding or menu contribution.  If you want to go down that route let me know.

**Comment by kmun** [comment]
Yes,  thanks.  I do you the X and Y cuts all the time.  I actually want to see all the traces as different colors.

Thanks,

Kam

            if (this.properties.draw_center_line) {


                ctx.lineWidth = (this.properties.center_line_style.lineWidth !== undefined) ? this.properties.center_line_style.lineWidth : 1;


                ctx.lineCap = (this.properties.center_line_style.lineCap !== undefined) ? this.properties.center_line_style.lineCap : "square";


                ctx.strokeStyle = (this.properties.center_line_style.strokeStyle !== undefined) ? this.properties.center_line_style.strokeStyle : Mx.fg;


                if (this.dragging || this.properties.highlight) {


                    ctx.lineWidth = Math.ceil(ctx.lineWidth * 1.2);


                }


                if (this.properties.direction === "vertical") {


                    ctx.beginPath();


                    ctx.moveTo(this.properties.center_location + 0.5, Mx.t);


                    ctx.lineTo(this.properties.center_location + 0.5, Mx.b);


                    ctx.stroke();


                } else if (this.properties.direction === "horizontal") {


                    ctx.beginPath();


                    ctx.moveTo(Mx.l, this.properties.center_location + 0.5);


                    ctx.lineTo(Mx.r, this.properties.center_location + 0.5);


                    ctx.stroke();


                }


            }


        }

**Comment by kmun** [comment]
Thanks,  I just coded it up.
